src/api/java/com/minecolonies/api/quests/IDialogueObjective.java

```diff
@@ -0,0 +1,136 @@
+package com.minecolonies.api.quests;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.minecolonies.api.IMinecoloniesAPI;
+import net.minecraft.network.chat.Component;
+import net.minecraft.resources.ResourceLocation;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.minecolonies.api.util.constant.QuestParseConstant.*;
+
+/**
+ * Dialogue type of objective interface.
+ */
+public interface IDialogueObjective extends IQuestObjective
+{
+
+    /**
+     * Getter for the dialogue tree.
+     * @return the tree.
+     */
+    DialogueElement getDialogueTree();
+
+
+    /**
+     * A dialogue element in the dialogue objective.
+     */
+    class DialogueElement implements IAnswerResult
+    {
+        /**
+         * The text the participant says.
+         */
+        private final String text;
```

Comment:
I don't mean for translation constants specifically, but rather for styled text and item links.  (You know, how you can put an `[Item]` in chat and it has a tooltip showing the item details.  Would be especially important if at some point someone wants to make a "get me this specific DO block" quest...)

Having said that, at least for quests included in mods rather than standalone it would be trivial to support translation as well -- just specify a translation key and use placeholders for the entity names.  This will work if the text is just hard-coded too, since it automatically uses it as a literal if the key isn't found as is.

There even is a syntax for doing entity name replacements already, although it's probably a bit too awkward and verbose to want to write into the JSON manually.  But with a datagen though.... 😁 (Pity you can't add custom contents to components though.)

A datagen would help with "spreading translation keys over files" too, at least at the source level, similar to how the research system does it.  Alternatively you could have a map of translations directly within the quest files, but then you'd have to implement the lookup yourself since it's non-standard (and it will be trickier to get it to work properly since you have to do the translation client-side).